scripts/lib/CIME/SystemTests/system_tests_compare_two.py

```diff
@@ -161,28 +193,38 @@ def run_phase(self, success_change=False):  # pylint: disable=arguments-differ
         Runs both phases of the two-phase test and compares their results
         If success_change is True, success requires some files to be different
         """
+        first_phase = self._case1.get_value("RESUBMIT") == 1
+        run_type = self._case1.get_value("RUN_TYPE")
 
         # First run
-        logger.info('Doing first run: ' + self._run_one_description)
-        self._activate_case1()
-        run_type = self._case1.get_value("RUN_TYPE")
-        self.run_indv(suffix = self._run_one_suffix)
+        if not self._multisubmit or first_phase:
```

Comment:
You have introduced a fair amount of new complexity into run_phase. That's fine, but I'd like to see some extra tests covering this new stuff. Specifically (in test_system_tests_compare_two.py):

(1) I'd like to see `test_run_phase_internal_calls` extended so that it also checks that `_case_one_custom_prerun_action`, `_case_one_custom_postrun_action` and similarly for case_two are called at the right points in the sequence. I'm confident that the current code is correct in this respect, but I'd like to see this added to cover any future refactorings.

(2) I'd like to see one or more tests covering the case where multisubmit=True. Possibly this could be done via some new tests similar to `test_run_phase_internal_calls`, one with RESUBMIT=1 and one with RESUBMIT=0 -- making sure the right things do and don't happen in each case. Maybe there's something more clever that could be tested, too, though nothing pops to mind. As in (1), this is less for the sake of testing the correctness of the current code, and more for the sake of making sure that this remains correct under future refactorings, since we otherwise don't have any tests that test this code.